bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,506 @@
+                continue
+
+            # Creating a line for the unit
+            if unit not in rhyme_track:
+                new_line = await self._init_unit(
+                    rhyme_track, unit, is_last_line
+                )
+                lines.append(new_line)
+            else:
+                new_line = await self._get_rhyming_line(
+                    word_rhymes=rhyme_track[unit],
+                    existing_lines=lines,
+                    error_info=(ctx, is_first_error, scheme, unit_count,
+                                time_start)
+                )
+                lines.append(new_line)
+
+                # If last line, it will not be referred to again
+                if not is_last_line:
+                    rhyme_track[unit] |= await self._get_rhyme_set(
+                        instance=self,
+                        word=self._get_last_word(new_line)
+                    )
+
+            unit_count[unit] -= 1
+
+        return lines
+
+    async def send_markov_poem(
+        self,
+        ctx: commands.Context,
+        scheme: str,
+        unit_count: Dict[str, bool],
+        time_start: datetime.time,
+        is_first_error: bool = True
+    ) -> None:
+        """Gives the user a paginated embed of their markov poem."""
+        lines = await self._get_generated_markov_poem(
+            instance=self,
+            timeout=self.POEM_TIMEOUT,
+            ctx=ctx,
+            scheme=scheme,
+            unit_count=unit_count,
+            time_start=time_start,
+            is_first_error=is_first_error
+        )
+
+        elapsed_time = datetime.now() - time_start
+        elapsed_time = elapsed_time.seconds
+
+        poem_embed = Embed(
+            title="A Markov Poem For " + str(ctx.author.name),
+            color=Colours.pink
+        )
+
+        await LinePaginator.paginate(
+            lines,
+            ctx,
+            poem_embed,
+            max_lines=31,
+            max_size=1000,
+            empty=False,
+            footer_text=f"Elapsed time: {elapsed_time}s\n"
+                        "Rhymes API provided by datamuse."
+        )
+
+    @commands.command(
+        aliases=("poem", "mpoem", "m_poem", "markpoem", "mark_poem"),
+        help=f"""
+            This command generates a poem via a markov chain, and all you have
+            to do is provide a rhyme scheme.
+
+            The rhyme scheme is made from characters separated by slashes. E.g
+            "abab/cdcd/efef/gg". The slashes denote a new stanza, i.e, they
+            create an empty line. Same characters mean that the lines rhyme.
+            Note that the characters are case sensitive! For example, a and A
+            represent two different rhyme schemes.
+
+            You may also use our existing rhyme scheme templates:
+            {chr(10).join((k + " --- " + v) for k, v in templates.items())}
+            """,
+        brief="Gives the user a love poem made with a markov chain."
+    )
```

Comment:
```suggestion
    @commands.command(
        aliases=("poem", "mpoem", "m_poem", "markpoem", "mark_poem"),
        help=f"""This command generates a poem via a markov chain, and all you have
            to do is provide a rhyme scheme.

            The rhyme scheme is made from characters separated by slashes. E.g
            "abab/cdcd/efef/gg". The slashes denote a new stanza, i.e, they
            create an empty line. Same characters mean that the lines rhyme.
            Note that the characters are case sensitive! For example, a and A
            represent two different rhyme schemes.

            You may also use our existing rhyme scheme templates:
            {chr(10).join((k + " --- " + v) for k, v in templates.items())}""",
        brief="Gives the user a love poem made with a markov chain."
    )
```

**Earlier:**

![Screenshot from 2021-02-14 14-35-34](https://user-images.githubusercontent.com/69356296/107872827-1be3c980-6ed3-11eb-8434-b99b90175034.png)

**Now:**

![Screenshot from 2021-02-14 14-44-30](https://user-images.githubusercontent.com/69356296/107872837-3158f380-6ed3-11eb-9a44-9fdee9e60b05.png)

Reply:
I'd prefer moving this to docstring.

Reply:
> I'd prefer moving this to docstring.

Here it shows the docstring which is useful for community members when they see the `help` on this command, but in the docstring it tells about the code which would be useful for contributors when they look at this file.

Reply:
discord.py is parsing command help from docstring